nvflare/mt/job_def_manager.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import pathlib
+import time
+import uuid
+from typing import Any, Dict, List
+
+from nvflare.apis.job_def_manager_spec import JobDefManagerSpec, RunStatus
+from nvflare.apis.storage import StorageSpec
+from nvflare.apis.study_manager_spec import StudyManagerSpec
+from nvflare.mt.job_def import JobMetaKey, Job, job_from_meta
+
+
+class _JobChecker:
+    def check_job(self, meta: dict) -> bool:
+        pass
+
+
+class _StatusChecker(_JobChecker):
+    def __init__(self, status_to_check):
+        self.result = []
+        self.status_to_check = status_to_check
+
+    def check_job(self, meta: dict):
+        if meta[JobMetaKey.STATUS] == self.status_to_check:
+            self.result.append(meta)
+        return True
```

Comment:
These checker looks more like a filter?
The "check_job" always return True.
In this case should we rename them?